chain/chain/src/resharding.rs

```diff
@@ -107,24 +121,34 @@ impl Chain {
     fn build_state_for_split_shards_impl(
         state_split_request: StateSplitRequest,
     ) -> Result<HashMap<ShardUId, StateRoot>, Error> {
-        let StateSplitRequest {
-            runtime_adapter,
-            shard_uid,
-            state_root,
-            next_epoch_shard_layout,
-            ..
-        } = state_split_request;
-        // TODO(resharding) use flat storage to split the trie here
-        let tries = runtime_adapter.get_tries();
+        let StateSplitRequest { tries, shard_uid, state_root, next_epoch_shard_layout, .. } =
+            state_split_request;
+
+        let store = tries.get_store();
         let trie = tries.get_view_trie_for_shard(shard_uid, state_root);
+        let mut iter = store_helper::iter_flat_state_entries(shard_uid, &store, None, None).map(
+            |entry| -> (Vec<u8>, Vec<u8>) {
+                let (key, value) = entry.unwrap();
+                let value = match value {
+                    FlatStateValue::Ref(ref_value) => {
+                        trie.storage.retrieve_raw_bytes(&ref_value.hash).unwrap().to_vec()
+                    }
+                    FlatStateValue::Inlined(inline_value) => inline_value,
+                };
+                (key, value)
+            },
+        );
+
         let shard_id = shard_uid.shard_id();
         let new_shards = next_epoch_shard_layout
             .get_split_shard_uids(shard_id)
             .ok_or(Error::InvalidShardId(shard_id))?;
         let mut state_roots: HashMap<_, _> =
             new_shards.iter().map(|shard_uid| (*shard_uid, Trie::EMPTY_ROOT)).collect();
+
+        // function to map account id to shard uid in range of child shards
         let split_shard_ids: HashSet<_> = new_shards.into_iter().collect();
-        let checked_account_id_to_shard_id = |account_id: &AccountId| {
+        let checked_account_id_to_shard_uid = |account_id: &AccountId| {
```

Comment:
nit: would be nice to move to a helper method

Reply:
Similarly here, we use captured variable `split_shard_ids` within the body. Just easier having it as lambda fn.

Reply:
Actually, managed to shift this to a helper...